CMakeLists.txt

```diff
@@ -1,73 +1,72 @@
-cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR) # Policies <= CMP0097 default to NEW
 
-message(AUTHOR_WARNING "If the build fails, please try the autotools/qmake method.")
-
-if (POLICY CMP0074)
-    cmake_policy(SET CMP0074 NEW)
-endif()
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
 
 # TODO: fix the macOS bundle stuff
-if (POLICY CMP0080)
-    cmake_policy(SET CMP0080 OLD)
+cmake_policy(SET CMP0080 OLD)
+
+# CMAKE_MSVC_RUNTIME_LIBRARY must be defined before project() or enable_language() to take effect
+if (MSVC)
+        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>$<$<BOOL:${MSVC_RUNTIME_DYNAMIC}>:DLL>")
 endif()
 
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
+# project version
 include(FunctionReadVersion)
+read_version("${CMAKE_CURRENT_SOURCE_DIR}/version.pri" QBT_VER_MAJOR QBT_VER_MINOR QBT_VER_BUGFIX QBT_VER_BUILD QBT_VER_STATUS)
+message(STATUS "Project version is: ${QBT_VER_MAJOR}.${QBT_VER_MINOR}.${QBT_VER_BUGFIX}.${QBT_VER_BUILD} (${QBT_VER_STATUS})")
```

Comment:
Consider moving that below replacing with `message(STATUS "Project version is: ${PROJECT_VERSION}"). Will be shorter.

Reply:
@zeule 
They are slightly different. The `$PROJECT_VERSION` cannot include `(${QBT_VER_STATUS})`. To be fair, this is something I would change about qBittorrent, along with the `QBT_VERSION_2`/`QBT_VERSION` "duplicate" defines, but that's not up to me, at least not in this PR.